plugins/utils/navbuddy.nix

```diff
@@ -0,0 +1,175 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.navic;
+  helpers = import ../helpers.nix {inherit lib;};
+  mkListStr = helpers.defaultNullOpts.mkNullable (types.listOf types.str);
+in {
+  options.plugins.navic =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "nvim-navbuddy";
+
+      package = helpers.mkPackageOption "nvim-navbuddy" pkgs.vimPlugins.nvim-navbuddy;
+
+      window = {
+        border = helpers.defaultNullOpts.mkBorder "rounded" "double" "solid" "none" ''
+          "rounded", "double", "solid", "none"  or an array with eight chars building up the border in a clockwise fashion
+          starting with the top-left corner. eg: { "╔", "═" ,"╗", "║", "╝", "═", "╚", "║" }.
+        '';
+
+        # TODO: how to do attribute set here or string ?
+        size = helpers.defaultNullOpts.mkStr "60%" ''
```

Comment:
how to do attribute set here or string ?

Reply:
First, at the very top of your module, I would add:
```nix
  percentageType = types.ints.between 0 100;
  mkPercentageOpt = default: helpers.defaultNullOpts.mkNullable percentageType (toString) default;
```

```nix
size =
  helpers.defaultNullOpts.mkNullable
  (with types;
    either
      percentageType
      (submodule {
        options = {
          height = mkPercentageOpt 40 "The height size (in %).";

          width = mkPercentageOpt 100 "The width size (in %).";
        };
      })
    )
    "60"
    "The size of BLABLABLA."
    ;
```

(same for `position`...)

Reply:
The, in the `config` part:
```nix
let
  toStrPercentage = value: "${value}%";
in
...

size = helpers.ifNonNull' size
  (
    if isInt size
    then toStrPercentage size
    else
    with size; {
      height = toStrPercentage height;
      width = toStrPercentage width;
    }
  ;
```

Reply:
statix seemed to be complaining about the `default`in `default: helpers.defaultNullOpts.mkNullable percentageType (toString) default;`. So I removed it :sweat: and it worked but I assume that is used to set the default value of say 60 ?

Reply:
Yes I made a mistake: `mkPercentageOpt = default: helpers.defaultNullOpts.mkNullable percentageType (toString default);